lib/puppet/util/log/destinations.rb

```diff
@@ -91,17 +91,24 @@ def initialize(path)
       end
     end
 
-    file = File.open(path,  File::WRONLY|File::CREAT|File::APPEND)
-    file.puts('[') if need_array_start
-
-    # Give ownership to the user and group puppet will run as
-    if Puppet.features.root? && !Puppet::Util::Platform.windows? && !file_exists
-      begin
-        FileUtils.chown(Puppet[:user], Puppet[:group], path)
-      rescue ArgumentError, Errno::EPERM
-        Puppet.err _("Unable to set ownership to %{user}:%{group} for log file: %{path}") % { user: Puppet[:user], group: Puppet[:group], path: path }
+    file = File.open(path, File::WRONLY|File::CREAT|File::APPEND)
+    if Puppet::Util::Platform.windows?
+      require 'puppet/util/windows'
+      dacl = Puppet::Util::Windows::AccessControlList.new
+      [
+        Puppet::Util::Windows::SID::LocalSystem,
+        Puppet::Util::Windows::SID::BuiltinAdministrators,
+        Puppet::Util::Windows::SID.name_to_sid(Puppet::Util::Windows::ADSI::User.current_user_name)
+      ].uniq.map do |sid|
+        dacl.allow(sid, FULL_CONTROL)
       end
+      sd = Puppet::Util::Windows::Security.get_security_descriptor(path)
+      new_sd = Puppet::Util::Windows::SecurityDescriptor.new(sd.owner, sd.group, dacl, true)
+      Puppet::Util::Windows::Security.set_security_descriptor(path, new_sd)
+    else
+      file.chmod(0640)
```

Comment:
This logic belongs in Puppet::FileSystem and there is already logic there to set permissions in the `replace_file` method.